### DISCUSSION TITLE
Clarifications

### DISCUSSION THREAD

**Discussion by GitTom** [jan-bogaerts/mark#5] [Q&A] [answered]
This looks very interesting, thanks.  Some feedback/questions...

The filename of the download for 0.0.2 indicates 0.1.1.  Perhaps that is a mistake?

To clarify ... MarkdownCode is a python app that converts md specs to js apps.   There is a also a sample generated app that is a js/electron based markdown editor.  But when I look at releases I see references to it being an electron app, so I guess the releases/downloads are actually for the sample?  What if I'm just interested in MarkdownCode: the js generator?

(I haven't actually tried it yet - I always take an extra pause before running a new app on my computer.)

The sample project (ie. the spec for the markdown app) is pretty complicated.  Maybe it would make the project more approachable if there was also a spec for a more conventional and simple web project?

I think it is common when looking at a project like this to be curious about the prompts used.  Perhaps you could note where to find them?

Have you tried MarkdownCode with GPT-4-1106-preview yet?  As I understand it there is no improvement in coding ability, but I guess it would have the advantage of knowledge of more current js/node world.

Sorry, don't mean to be demanding - please just take these as feedback/thoughts.

**Comment by jan-bogaerts** [comment, marked as answer]
Hi Tom,
Apologies for the confusion. Let me try to explain it a little better (the frontpage is perhaps still a little confusing, it was actually originally generated with gpt 3.5, way in the beginning, before I even properly knew myself what direction it would go in to. I've tried updating it a little since. Not there yet...

> MarkdownCode is a python app that converts md specs to js apps.

This is how it originally started out, yes, though it was always the plan to let the tool evolve to a js app. The very first version was written in python, has a very limited set of features and is fairly demanding to set up, though the prompts used have remained the same (mostly). I am currently still using the python version extensively while working on the next version.
The current set of prompts that are used in the app, are indeed for generating javascript/typescript apps (web/electron). The python version also uses a fixed set of prompts, but the new electron version dynamically loads the prompts as plugins, so it should be able to do a lot more eventually then just render js apps.
Another set of prompts that I am currently thinking of, would be to convert older angular apps into react (something I could use for work), similarly, a set of prompts to split up very very large code files would also be useful for work...

> There is a also a sample generated app that is a js/electron based markdown editor. 

The sample is actually the next version of the tool. It's an editor, but most importantly:
- it's a db that stores the results of the prompts and tracks in a clever way when results have become out of date and informs you using various state icons. (ex, when a line is changed, it rebuilds all the inputs that were dependent on that line and checks if the input of the prompt has changed.
- It manages and dynamically loads all the prompts that are available in the tool.  The only prompts that are baked in to the app, are used to build plugins. (ex: https://github.com/jan-bogaerts/markdownCode/blob/main/src/markdownCode.md#plugin-renderer-service) Note: this is a text fragment out of the full application, plugins can be written simpler, see further down)
- I'm hoping to also be able to soon add more input features like screen capture (send an image of the app to a prompt), file-monitoring (update a description when a file is manually changed),...

> The sample project (ie. the spec for the markdown app) is pretty complicated. Maybe it would make the project more approachable if there was also a spec for a more conventional and simple web project?

Yes, you are completely correct, it really needs a simple sample  app, just haven't gotten there yet. 

> I think it is common when looking at a project like this to be curious about the prompts used. Perhaps you could note where to find them?

take a look at this: https://github.com/jan-bogaerts/markdownCode/blob/main/src/plugins.md This file contains the specs for the prompts that will be used to build the app itself. You'll notice a lot of python code in there, that's because these prompts are just the converted python prompts that you can find here:  https://github.com/jan-bogaerts/markdownCode/tree/main/scripts  (some of the more important ones are 'render_component' and 'render_class'

> Have you tried MarkdownCode with GPT-4-1106-preview yet? As I understand it there is no improvement in coding ability, but I guess it would have the advantage of knowledge of more current js/node world.

yes. the engine allows you to select which models to use by which prompts (can even be overwritten for each text fragment individually). In the python version, this was by editing a json file. in the electron app, you can still edit a json file or use the interface to select the models...
Personally, I haven't yet seen much difference yet. the quality of the models varies at times, but some things remain difficult.
No matter what I tried, it still wont render the markdown parser code just on a description only. In fact, it wont generate it in 1 file, even if most of it is supplied as python code.

hope this clears things up some more? I'm planning to do a new release in the coming weeks, which should be a true alpha version of the tool, but as this is a hobby project, time schedules can vary sometimes...

**Comment by GitTom** [comment]
Thanks for the clear and detailed answers.  Yes, that really helps!

It sounds great - your vision for this tool seems very interesting and promising.

But it also sounds like it is very much in flux at the moment, so I'm better to wait for the next version before I give it a try.
